### DISCUSSION TITLE
Should schema.rs be commited to source control?

### DISCUSSION THREAD

**Discussion by meclondrej** [diesel-rs/diesel#4248] [Polls]
Should i put it in my .gitignore? I know that according to the "don't commit generated code to source control" philosophy I shouldn't, but that definitely has downsides, eg. someone who clones the repo cannot compile without generating the schema with the CLI.

**Comment by weiznich** [comment]
I usually advice to checkin the `schema.rs` file into your VCS. You should then setup a CI job that uses `diesel migration run --locked-schema` to check if the schema is up to date with the provided migrations. In that way you decouple the actual build process to from having access to a database.
